### ISSUE TITLE
[breaking?] Replace ErrorException for not-implemented

### ISSUE THREAD

**Issue by matbesancon** [JuliaGraphs/LightGraphs.jl#1218]
`ErrorException` is too generic for not-implemented interface, `_NI("func")` should either throw a method error or a custom `Exception` type

**Comment by matbesancon** [comment]
The breaking is question-marked since it changes the interface in a breaking way if somebody is expecting an `ErrorException` when not implementing

**Comment by simonschoelly** [comment]
One option would be to just implement `function f end`. Then the symbol can still be exported, and shows `f (generic function with 0 methods)` in the REPL. Docstrings can also be attached.

The disadvantage of this is, that the information on the parameters get lost. We could get around this, by having a macro `@NI` or `@Interface` that takes a function declaration and strips the parameters off. Not sure if that would work with docstrings then.
So you would write for example
```
 '''
  vertices(g)
Return the vertices of.
 '''
@NI function vertices(g::AbstractGraph) end
```

**Comment by sbromberger** [comment]
I really prefer having an exception for this, as it means that the graph has not been specified adequately to conform to the interface.

**Comment by matbesancon** [comment]
That's why I was suggesting creating a new exception type `GraphInterfaceException` or such, and still throw it.

**Comment by sbromberger** [comment]
Yes - I agree with you.

I do not view this as breaking since we're still throwing an exception, but I'd like to defer to @jpfairbanks who has a better understanding of SemVer than I do.

**Comment by matbesancon** [comment]
nope, I'll do it soon

**Comment by matbesancon** [comment]
fixed by #1245
